packages/blockchain-tools/src/interfaces/Web3Server.ts

```diff
@@ -315,24 +325,33 @@ export default class Web3Server extends ChainBaseClass {
   //   }
   //   return result
   // }
-  async getTransactionInfo(txId: string) {
-    const txInfo = await this.fetchLedger("eth_getTransactionByHash", [txId])
-    console.log('TX', txInfo)
-
-    if (!txInfo || txInfo.error) {
-      return { error: "Transaction not found" }
-    }
-
-    return {
-      id: txInfo.hash,
-      hash: txInfo.hash,
-      from: txInfo.from,
-      to: txInfo.to,
-      value: txInfo.value,
-      fee: (BigInt(txInfo.gas) * BigInt(txInfo.gasPrice)).toString(),
-      blockNumber: Number.parseInt(txInfo.blockNumber, 16),
-      timestamp: "" // Ethereum transactions do not directly provide a timestamp
+  async getTransactionInfo(txId: string, wait?:boolean) {
+    const secs = [2,2,3,3,4,4,5,5,6,6]
```

Comment:
What's the long term solution for this? Do we need to use a different package that has web sockets or something?

Reply:
Some SDKs have a submitAndWait method, some don't. Sometimes waiting a fixed amount of time is not enough, so we use incremental waiting time with a timeout of 40 secs. Sure there must be a library somewhere that implements the same idea under the hood